fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java

```diff
@@ -133,7 +135,6 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
                 LOG.warn("fn {} not exists", this.toSqlImpl());
                 throw new AnalysisException(getFunctionNotFoundError(collectChildReturnTypes()));
             }
-            fn.setReturnType(getChild(0).getType());
         } else if (fnName.getFunction().equalsIgnoreCase("array_filter")) {
```

Comment:
why remove this line?

Reply:
The return type has already been specified in the `fn` when we create function in `doris_builtins_functions.py`. In this case, the return type of `fn` is `Int64`, but the the return type of children 0 is `Array(Bool)`, we shouldn't change the return type of the `fn`.